src/raftstore/store/peer.rs

```diff
@@ -898,12 +871,11 @@ impl Peer {
             self.pending_reads.clear_uncommitted(term);
         }
 
-        if let Some(Either::Right(_)) = self.leader_lease_expired_time {
-            return;
-        }
-
         if let Some(propose_time) = propose_time {
-            self.update_lease_with(propose_time);
+            if self.leader_lease.inspect(Some(propose_time)) == LeaseState::Suspect {
```

Comment:
Is `propose_time` appropriate here?

Reply:
I think it okay, `propose_time` is a place holder, here cares about `Suspect` only,
and if it is in `Suspect` phase, the actual timestamp is useless.